pkg/cmd/roachtest/spec/cluster_spec.go

```diff
@@ -153,12 +153,46 @@ func MakeClusterSpec(nodeCount int, opts ...Option) ClusterSpec {
 
 // ClustersCompatible returns true if the clusters are compatible, i.e. the test
 // asking for s2 can reuse s1.
-func ClustersCompatible(s1, s2 ClusterSpec) bool {
-	s1.Lifetime = 0
-	s2.Lifetime = 0
+func ClustersCompatible(s1, s2 ClusterSpec, cloud string) bool {
+	// only consider the specification of the cloud that we are running in
+	clearClusterSpecFields(&s1, cloud)
+	clearClusterSpecFields(&s2, cloud)
 	return s1 == s2
 }
 
+func clearClusterSpecFields(c *ClusterSpec, cloud string) {
+	c.Lifetime = 0
+	switch cloud {
+	// only consider the specification of the cloud that we are running in
+	case GCE:
```

Comment:
See my suggestion around how we can avoid clearing "other" clouds using reflection.